crates/evm/src/evm/db.rs

```diff
@@ -121,6 +121,32 @@ impl<C: sov_modules_api::Context> Database for EvmDb<'_, C> {
     }
 }
 
+/// A trait to check if an account is newly created.
+/// This is useful when calculating diff size for a transactions
+pub trait IsAccountNewlyCreatedProvider {
+    /// Check if an account is newly created
+    /// By querying `Evm::account_exists`
+    fn is_account_new(&mut self, address: &Address) -> bool;
+}
+
+impl<C: sov_modules_api::Context> IsAccountNewlyCreatedProvider for EvmDb<'_, C> {
+    fn is_account_new(&mut self, address: &Address) -> bool {
```

Comment:
How it can be new if it does not exist? This semantics `new/old` is "artificial" and exists only in state diff calc.